File: examples/governance_static_go/domain/recorded_metric.go
package domain

import "time"

type RecordedMetric struct {
	AggregateID string    `json:"aggregate_id"`
	AggregateId string `json:"aggregate_id"`
	ModelId string `json:"model_id"`
	DeploymentId string `json:"deployment_id"`
	MetricName string `json:"metric_name"`
	Value float64 `json:"value"`
	Threshold float64 `json:"threshold"`
	RecordedAt time.Time `json:"recorded_at"`
	OccurredAt time.Time `json:"occurred_at"`
}

func (e RecordedMetric) EventName() string { return "RecordedMetric" }

func (e RecordedMetric) GetOccurredAt() time.Time { return e.OccurredAt }

File: examples/governance/operations_domain/lib/operations_domain/monitoring/events/recorded_metric.rb
module OperationsDomain
  class Monitoring
    module Events
      class RecordedMetric
        attr_reader :aggregate_id, :model_id, :deployment_id, :metric_name, :value, :threshold, :recorded_at, :occurred_at

        def initialize(aggregate_id: nil, model_id: nil, deployment_id: nil, metric_name: nil, value: nil, threshold: nil, recorded_at: nil)
          @aggregate_id = aggregate_id
          @model_id = model_id
          @deployment_id = deployment_id
          @metric_name = metric_name
          @value = value
          @threshold = threshold
          @recorded_at = recorded_at
          @occurred_at = Time.now
          freeze
        end
      end
    end
  end
end
